tests/python/unittest/common.py

```diff
@@ -94,6 +95,23 @@ def random_seed(seed=None):
         random.seed(next_seed)
 
 
+def assert_raises_cudnn_disabled(assertion_error=False):
+    def test_helper(orig_test):
+        @make_decorator(orig_test)
+        def test_new(*args, **kwargs):
+            cudnn_disabled = (os.getenv('CUDNN_DISABLED') == "ON")
+            if not cudnn_disabled or mx.context.current_context().device_type == 'cpu':
+                orig_test(*args, **kwargs)
+            else:
+                if assertion_error:
```

Comment:
When do we want to differentiate between assertion errors? I think we should always handle them, no matter what.

Reply:
this is to differentiate between failures and the ops that dont support cudnn disabled. for example some ops when used with cudnn disabled would just throw an exception saying not supported. others would proceed but would compute wrongly and throw an assertion error.

Reply:
I see, thanks for elaborating. I'm afraid that this could be a bit error prone in it's usage.

 Also, I don't think this is what should happen. If a user uses an operator although it's not supported, it should not make a random false calculation but rather throw the error properly. This method would mask a problem that a user would silently encounter. We should rather investigate why the error is not being thrown and fix that.

Reply:
Also, is there no CPU implementation of these operators we could fall back to instead?

Reply:
yep the issue was uncovered as part of the PR and is documented here: https://github.com/apache/incubator-mxnet/issues/11568 . As I have mentioned in #11470, this is not something we should be blocking the PR for. This issue seems could have been there for a long time. Its possible that there are user codes depending on the buggy behavior and raising an exception would be a breaking change for these users. Its better to fix the issue with cudnn enabled rather than throwing an exception saying cudnn not supported atleast for 1.3.

Reply:
Agree, let's not block that PR. But please add a comment here that this is a temporary solution and link it with the issue. 

Would you mind if we start a thread on dev@ to discuss this topic? I think we should have a thorough analysis on this topic. I agree that this might cause a breaking change, but on the other hand, our users might have worked with wrong operators the entire time which might have caused unknown side effects - something that's probably even worse than a detailed error message.

Reply:
right now the only op that is affected is `mx.sym.SpatialTransformer`. If cudnn is enabled `CuDNNSpatialTransformerOp` is used. If it is not enabled then the mxnet implementation for the op is used. Why this implementation is incorrect needs to be investigated. Do you think we need to start a discussion for only this op ?

Reply:
If it is only that operator, I'd rather say that we should deactivate that test and Mark it as flaky. We can then remove this special catch for assertion errors.

In that case we don't need a discussion.

Reply:
we need to still test it when cudnn is enabled.

Reply:
We currently have a 0 tolerance policy for faulty tests and deactivate them temporarily while taking reduced test coverage as a pill we have to swallow. It is a release requirement that all tests have to be enabled, so don't worry: it won't stay disables for long.

Reply:
why disable the test when it can be avoided. If it is for visibility then we already have an open issue. we can also call it a disabled test since it is disabled for cudnn off.

Reply:
Sure, as you prefer. Just wanted to avoid having this hack in there, but I don't mind. Could you please add a comment in that test, linking to the issue and stating that the test is disabled when running without cudnn?

Reply:
will do